### ISSUE TITLE
Problem compiling tensorflow with SyCL support

### ISSUE THREAD

**Issue by fvella** [codeplaysoftware/tensorflow#7]
Hi, I have a problem compiling TensorFlow with SyCL support.
I am following the instructions listed [here](https://developer.codeplay.com/computecppce/latest/tensorflow-arm-setup-guide)
It looks like to be related to Eigen3 and Sycl support. 

### System information
== cat /etc/issue ===============================================
Linux hikey961 4.4.74-00216-g10816f6 #3 SMP PREEMPT Thu Jul 6 14:38:42 BST 2017 aarch64 GNU/Linux
VERSION_ID="9"
VERSION="9 (stretch)"

== are we in docker =============================================
No

== compiler =====================================================
c++ (Debian 6.3.0-18+deb9u1) 6.3.0 20170516
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

== uname -a =====================================================
Linux hikey961 4.4.74-00216-g10816f6 #3 SMP PREEMPT Thu Jul 6 14:38:42 BST 2017 aarch64 GNU/Linux

== check pips ===================================================
numpy (1.14.0)
protobuf (3.4.0)

== check for virtualenv =========================================
False

== tensorflow import ============================================
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "tensorflow/python/__init__.py", line 49, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "tensorflow/python/pywrap_tensorflow.py", line 25, in <module>
    from tensorflow.python.platform import self_check
ImportError: No module named platform

== env ==========================================================
LD_LIBRARY_PATH /usr/lib/aarch64-linux-gnu//mali/fbdev
DYLD_LIBRARY_PATH is unset

== nvidia-smi ===================================================
tools/tf_env_collect.sh: line 105: nvidia-smi: command not found

== cuda libs  ===================================================

### Describe the problem
Error Compiling  tensorflow following the instructions reported https://developer.codeplay.com/computecppce/latest/tensorflow-arm-setup-guide

### Source code / logs
```ERROR: /home/flavio/CK_TOOLS/tensorflow-opencl-4aff06b-lib.armcl-18.03-e40997bb-linux-64/src/tensorflow/core/kernels/BUILD:210:1: C++ compilation of rule '//tensorflow/
core/kernels:initializable_lookup_table' failed (Exit 1)                                  
In file included from tensorflow/core/kernels/initializable_lookup_table.cc:16:         
In file included from ./tensorflow/core/kernels/initializable_lookup_table.h:19:                                                                                        In file included from ./tensorflow/core/framework/lookup_interface.h:19:       
In file included from ./tensorflow/core/framework/resource_mgr.h:24:     
In file included from ./tensorflow/core/framework/common_shape_fns.h:20:
In file included from ./tensorflow/core/framework/shape_inference.h:20:                                                                                                 In file included from ./tensorflow/core/framework/node_def_util.h:23:
In file included from ./tensorflow/core/framework/attr_value_util.h:23:
In file included from ./tensorflow/core/framework/partial_tensor_shape.h:20:
In file included from ./tensorflow/core/framework/tensor_shape.h:21:
In file included from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:100:
external/eigen_archive/unsupported/Eigen/CXX11/src/Tensor/TensorDeviceSycl.h:344:12: error: call to constructor of 'accessor' (aka 'accessor<unsigned char, 1, (cl::sycl
::access::mode)3U, global_access>') is ambiguous
    return accessor(buffer, cgh, cl::sycl::range<1>(offset), cl::sycl::range<1>(n_bytes));
           ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/eigen_archive/unsupported/Eigen/CXX11/src/Tensor/TensorDeviceSycl.h:238:32: note: in instantiation of function template specialization 'Eigen::QueueInterface::
get_range_accessor<cl::sycl::access::mode::discard_write, unsigned long>' requested here
      write_accessor dst_acc = get_range_accessor<write_mode>(cgh, dst, n);
                               ^
external/local_config_sycl/crosstool/../sycl/include/SYCL/accessor.h:1443:3: note: candidate constructor [with allocatorT = cl::sycl::detail::aligned_mem::aligned_alloc
ator<unsigned char>, overloadDependantT = unsigned char, $2 = void]
  accessor(buffer<elemT, kDims, allocatorT> &bufferRef, handler &commandHandler```

**Comment by lukeiwanski** [comment]
Hi @fvella,

What ComputeCpp version do you have and what TF commit are you compiling?

Thanks,

**Comment by fvella** [comment]
Hi,
I am using ```TensorFlow 4aff06b``` from your fork and ```ComputeCPP 0.7.0 Ubuntu 16.04 ARM 64```

**Comment by lukeiwanski** [comment]
For that revision you need ComputeCpp 0.5.1

**Comment by fvella** [comment]
OK thanks. 
BTW What about TensorFlow master version and the latest ComputeCPP? Did you already test/benchmark?

**Comment by lukeiwanski** [comment]
It is on my TODO list.
Which TF version are you interested in?

**Comment by fvella** [comment]
I am interested in Tensorflow 1.8.

**Comment by lukeiwanski** [comment]
This repo is more conservative. We will get to 1.8 at some point.
I would recommend checking out my fork: https://github.com/lukeiwanski/tensorflow/tree/integration/1.8
`integration/1.8` branch is a "work-in-progress", but will run with ComputeCpp 0.7.0. It won't work with the recently-released 0.8.0. We haven't benchmarked it yet either, as we are still working on optimising the kernel code.

**Comment by lukeiwanski** [comment]
I believe this issue can be closed at this point.
